pkg/evpn/bridge.go

```diff
@@ -0,0 +1,225 @@
+	}
+	// use netlink to delete vxlan device
+	if err := netlink.LinkDel(vxlan); err != nil {
+		fmt.Printf("Failed to delete link: %v", err)
+		return nil, err
+	}
+	// remove from the Database
+	delete(s.Bridges, obj.Name)
+	return &emptypb.Empty{}, nil
+}
+
+// UpdateLogicalBridge updates a LogicalBridge
+func (s *Server) UpdateLogicalBridge(_ context.Context, in *pb.UpdateLogicalBridgeRequest) (*pb.LogicalBridge, error) {
+	log.Printf("UpdateLogicalBridge: Received from client: %v", in)
+	// check required fields
+	if err := fieldbehavior.ValidateRequiredFields(in); err != nil {
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	// Validate that a resource name conforms to the restrictions outlined in AIP-122.
+	if err := resourcename.Validate(in.LogicalBridge.Name); err != nil {
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	// fetch object from the database
+	volume, ok := s.Bridges[in.LogicalBridge.Name]
+	if !ok {
+		// TODO: introduce "in.AllowMissing" field. In case "true", create a new resource, don't return error
+		err := status.Errorf(codes.NotFound, "unable to find key %s", in.LogicalBridge.Name)
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	// update_mask = 2
+	if err := fieldmask.Validate(in.UpdateMask, in.LogicalBridge); err != nil {
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	resourceID := path.Base(volume.Name)
+	iface, err := netlink.LinkByName(resourceID)
+	if err != nil {
+		err := status.Errorf(codes.NotFound, "unable to find key %s", resourceID)
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	// base := iface.Attrs()
+	// iface.MTU = 1500 // TODO: remove this, just an example
+	if err := netlink.LinkModify(iface); err != nil {
+		fmt.Printf("Failed to update link: %v", err)
+		return nil, err
+	}
+	// TODO: replace cloud -> evpn
+	response := proto.Clone(in.LogicalBridge).(*pb.LogicalBridge)
+	response.Status = &pb.LogicalBridgeStatus{OperStatus: pb.LBOperStatus_LB_OPER_STATUS_UP}
+	s.Bridges[in.LogicalBridge.Name] = response
+	log.Printf("UpdateLogicalBridge: Sending to client: %v", response)
+	return response, nil
+}
+
+// GetLogicalBridge gets a LogicalBridge
+func (s *Server) GetLogicalBridge(_ context.Context, in *pb.GetLogicalBridgeRequest) (*pb.LogicalBridge, error) {
+	log.Printf("GetLogicalBridge: Received from client: %v", in)
+	// check required fields
+	if err := fieldbehavior.ValidateRequiredFields(in); err != nil {
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	// Validate that a resource name conforms to the restrictions outlined in AIP-122.
+	if err := resourcename.Validate(in.Name); err != nil {
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	// fetch object from the database
+	snet, ok := s.Bridges[in.Name]
+	if !ok {
+		err := status.Errorf(codes.NotFound, "unable to find key %s", in.Name)
+		log.Printf("error: %v", err)
+		return nil, err
+	}
+	resourceID := path.Base(snet.Name)
+	_, err := netlink.LinkByName(resourceID)
```

Comment:
What Link do we expect to get from Linux that has the resourceID name ?

Reply:
I assumed `Vxlan` here

Reply:
Sorry I didn't understand what you mean ? What is this VXLAN that you are talking about ?

Reply:
for example:
```
ip link add vxlan-<LB-vlan-id> type vxlan id <LB-vni> local <vtep-ip> dstport 4789 nolearning proxy
ip link set vxlan-<LB-vlan-id> master br-tenant up
bridge vlan add dev vxlan-<LB-vlan-id> vid <LB-vlan-id> pvid untagged
bridge link set dev vxlan-<LB-vlan-id> neigh_suppress on
```
in this case will be `vxlan-<LB-vlan-id>`